src/main/groovy/com/cloudbees/groovy/cps/CpsTransformer.groovy

```diff
@@ -78,7 +79,7 @@ import static org.codehaus.groovy.syntax.Types.*
  * @author Kohsuke Kawaguchi
  */
 class CpsTransformer extends CompilationCustomizer implements GroovyCodeVisitor {
-    private int iota=0;
+    private static final AtomicLong iota = new AtomicLong();
```

Comment:
@jglick Other possibility: overflowing this value. 1000 methods * 3000 pipeline jobs * 1000 executions > Integer.MAX_VALUE -- or is there something limiting this to prevent it exceeding that limit in reasonable cases?

Obviously that's a reasonable limit per-script, or per-job, but not necessarily globally.